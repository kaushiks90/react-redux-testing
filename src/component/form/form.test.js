import React from "react";
// import toJson from 'enzyme-to-json'
import {
  shallow
} from "enzyme";
import {
  findByTestAttr
} from "../../utils/index";
import Form from "./index";
import api from "./api";

const setup = (props = {}) => {
    const component = shallow( < Form {
        ...props
      }
      />);
      return component;
    };

    describe("Form Component", () => {
      let component;
      beforeEach(() => {
        component = setup();
      });
      it("should have the checkbox checked", () => {
        let wrapper = findByTestAttr(component, "checked");
        expect(wrapper.props().checked).toBe(true);
      });
      it("should allow user to input in the form", () => {
        const wrapper = shallow( < Form / > );
        wrapper
          .find(`[data-test='${"name"}']`)
          .simulate("change", {
            target: {
              name: "name",
              value: "cats"
            }
          });

        expect(wrapper.state("name")).toEqual("cats");
      });
      it("submits the form", () => {
        jest
          .spyOn(api, "addUser")
          .mockImplementation(() => Promise.resolve({
            data: "New User Added"
          }));
        const wrapper = shallow( < Form / > );
        wrapper
          .find(`[data-test='${"name"}']`)
          .simulate("change", {
            target: {
              name: "name",
              value: "cats"
            }
          });
        wrapper.find(`[data-test='${"email"}']`).simulate("change", {
          target: {
            name: "email",
            value: "fdgd@gmiail.com"
          }
        });
        wrapper
          .find(`[data-test='${"number"}']`)
          .simulate("change", {
            target: {
              name: "number",
              value: "546456456"
            }
          });
        wrapper
          .find(`[data-test='${"addUserForm"}']`)
          .simulate("submit", {
            preventDefault: () => {}
          });

        expect(api.addUser).toHaveBeenCalledWith(
          "cats",
          "fdgd@gmiail.com",
          "546456456"
        );
      });

      it('matches saved snapshot', () => {
        const wrapper = shallow( < Form / > )
        expect(wrapper).toMatchSnapshot()
      })
    });