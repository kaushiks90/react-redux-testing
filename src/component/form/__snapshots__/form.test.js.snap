// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component matches saved snapshot 1`] = `ShallowWrapper {}`;
